### PR TITLE
tdf: init at 0-unstable-2024-05-29

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
[tdf](https://github.com/itsjunetime/tdf) is a tui-based PDF viewer written in rust

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Gerg-L** [comment]
It's probably worth making an issue on upstream to add a license

**Comment by Gerg-L** [comment]
Yeah when asked usually people just slap a MIT on it

**Comment by Gerg-L** [comment]
@luftmensch-luftmensch license was added. mpl2.0

**Comment by luftmensch-luftmensch** [comment]
> @luftmensch-luftmensch license was added. mpl2.0

Applied changes!

File: pkgs/by-name/td/tdf/package.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+rustPlatform.buildRustPackage {
+  pname = "tdf";
+  version = "0-unstable-2024-05-29";
+
+  src = fetchFromGitHub {
+    owner = "itsjunetime";
+    repo = "tdf";
+    fetchSubmodules = true;
+    rev = "51b49b01115d9e25c597c3c29768bf1069a6c6b2";
+    hash = "sha256-CWnB1e/EWHqSqiOpK2SjlbAry5GHwPlZNP9RQkLA4zQ=";
```
Comment (Gerg-L) [inline pkgs/by-name/td/tdf/package.nix:20]:
```suggestion
    rev = "1e89a93879fbb7a5b78f43fbb61d0c99ce184e4f";
    hash = "sha256-Zl5XAaEGTPbOjlFltCQqhPyEExk7Xs+NY4uNtagCgCw=";
```
Update past the license being added then LGTM

Comment (luftmensch-luftmensch) [reply pkgs/by-name/td/tdf/package.nix]: Done!

**Comment by Gerg-L** [comment]
Result of `nixpkgs-review pr 315575` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tdf</li>
  </ul>
</details>

**Review by Gerg-L — APPROVED** [review]

File: pkgs/by-name/td/tdf/package.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    };
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    cairo
+    glib
+    poppler
+  ];
+
```
Comment (AndersonTorres) [inline pkgs/by-name/td/tdf/package.nix:38]: Have you tried `strictDeps = true;`?

Comment (luftmensch-luftmensch) [reply pkgs/by-name/td/tdf/package.nix:38]: Applied suggestion!

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**